extensions/amp-analytics/0.1/session-manager.js

```diff
@@ -76,41 +76,33 @@ export class SessionManager {
    * Get the value from the session per the vendor.
    * @param {string|undefined} type
    * @param {SESSION_VALUES} value
-   * @param {boolean=} opt_persist
    * @return {!Promise<number|undefined>}
    */
-  getSessionValue(type, value, opt_persist) {
-    return this.get(type).then((session) => {
-      if (value === SESSION_VALUES.EVENT_TIMESTAMP) {
-        return this.getSessionEventTimestamp_(type, session, opt_persist);
-      }
-      return session?.[value];
-    });
+  getSessionValue(type, value) {
+    return this.get(type).then((session) => session?.[value]);
   }
 
   /**
-   * Get the value from the session per the vendor.
+   * Update EventTimestamp for this session async.
+   * Passes in a callback to set the EventTimestamp,
+   * to be used when the session it retrieved or created.
    * @param {string} type
-   * @param {SessionInfoDef} session
-   * @param {boolean} persist
-   * @return {number|undefined}
+   * @return {!Promise}
    */
-  getSessionEventTimestamp_(type, session, persist) {
-    const lastTimestamp = session[SESSION_VALUES.EVENT_TIMESTAMP];
-    if (persist) {
-      this.sessions_[type] = this.updateSession_(session, true);
-      this.setSession_(type, session);
-    }
-    return lastTimestamp;
+  updateEventTimestamp(type) {
+    return this.get(type, (session) => {
```

Comment:
Suggestion: Consider using the same convention here (currently passing a callback) and above (promise chaining)

Reply:
Discussed offline, they're doing different things; i.e. this one is a callback.